projects/systelab-components/src/lib/directives/keyup-debounce.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { AfterContentInit, AfterViewInit, Directive, ElementRef, EventEmitter, HostListener, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+@Directive({
+	selector: '[keyup-debounce]'
+})
+export class KeyupDebounceDirective implements OnInit, OnDestroy {
+
+	@Input() public keyupDebounceTime = 350;
+	@Output() public keyupDebounced = new EventEmitter();
+	private debouncer = new Subject();
+	private subscription: Subscription;
+	public constructor(private el: ElementRef) {
+	}
+
+
+	public ngOnInit(): void {
+		this.subscription = this.debouncer
+			.pipe(debounceTime(this.keyupDebounceTime))
+			.subscribe(e => this.keyupDebounced.emit(e));
```

Comment:
I understand that the event that is being emitted here is the same one that has been passed to the debouncer inside the keyupEvent function. 
Is it ok that the the published event has the preventDefault() and the stopPropagation() method has been called on this event that will be emitted?

Reply:
Perfect, thanks! 👍